examples/jaeger-native-tracing/Dockerfile-frontenvoy

```diff
@@ -0,0 +1,8 @@
+FROM envoyproxy/envoy-dev:latest
+
+RUN apt-get update && apt-get -q install -y \
+    curl
+COPY ./front-envoy-jaeger.yaml /etc/front-envoy.yaml
+COPY ./libjaegertracing.so.0.4.2 /usr/local/lib/libjaegertracing_plugin.so
```

Comment:
Wait, sorry, I think we can do `ADD` or `curl`-ing in here?

Reply:
well, sort of. We can do `curl` + `tar`, not sure about `ADD` (which seems discouraged in this context - https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy)

i didnt do this because the source seemed a bit unofficial.

im happy to implement either the curl way or with binary in repo.

Ill add a comment in the recipe now about where the binary came from etc

Reply:
OK. I'll let @lizan to comment.

1. We can have that but under certain env var or commented.
2. Make sure we do checksum as well.

Reply:
@lizan do you have a preference re `jaeger-native-tracing` binary ?

should we:

- curl the binary from the `getenvoy` repo/file, checksum it and include
- or, just include the binary in the envoy repo

lmk and ill update the PR

Reply:
let's do curl with checksum